src/transformState.js

```diff
@@ -4,8 +4,27 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'clear':
+        for (const property in state) {
+          delete state[property];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const each of action.keysToRemove) {
+          delete state[each];
+        }
+        break;
```

Comment:
Don't forget to add a default case
![image](https://user-images.githubusercontent.com/96409563/210328542-dd6bdcb4-c502-4c98-a0b6-4ad6f659d14f.png)